pkg/api/v1/types.go

```diff
@@ -187,6 +187,11 @@ type ObjectMeta struct {
 	// from the list. If the deletionTimestamp of the object is non-nil, entries
 	// in this list can only be removed.
 	Finalizers []string `json:"finalizers,omitempty" patchStrategy:"merge" protobuf:"bytes,14,rep,name=finalizers"`
+
+	// The cluster which the object belongs to. The federation apiserver/controllers use this to
+	// distinguish the object at the federation level from it's constituents at the cluster level.
+	// An empty value means the object belongs to the local cluster (the one the client connects to).
+	Cluster string `json:"cluster,omitempty" protobuf:"bytes,15,opt,name=cluster"`
```

Comment:
Add "Cannot be updated" as we say for Name and Namespace.
Will also need to add logic to ensure that it is not updated.